modules/createBroadcast.js

```diff
@@ -0,0 +1,117 @@
+import React from "react"
+import PropTypes from "prop-types"
+import invariant from "invariant"
+
+function createBroadcast(initialValue) {
+  let subscribers = []
+  let currentValue = initialValue
+
+  const publish = value => {
+    currentValue = value
+    subscribers.forEach(subscriber => subscriber(currentValue))
+  }
+
+  const subscribe = subscriber => {
+    subscribers.push(subscriber)
+    return () => (subscribers = subscribers.filter(item => item !== subscriber))
+  }
+
+  let broadcastInstance = null
+
+  /**
+   * A <Broadcast> is a container for a "value" that its <Subscriber>
+   * may subscribe to. A <Broadcast> may only be rendered once.
+   */
+  class Broadcast extends React.Component {
+    /**
+     * For convenience when setting up a component that tracks this <Broadcast>'s
+     * value in state.
+     *
+     *     const { Broadcast, Subscriber } = createBroadcast("value")
+     *
+     *     class MyComponent {
+     *       state = {
+     *         broadcastValue: Broadcast.initialValue
+     *       }
+     *
+     *       // ...
+     *
+     *       render() {
+     *         return <Broadcast value={this.state.broadcastValue}/>
+     *       }
+     *     }
+     */
+    static initialValue = initialValue
+
+    componentDidMount() {
+      invariant(
+        broadcastInstance == null,
+        "You cannot render the same <Broadcast> twice! There must be only one source of truth. " +
+          "Instead of rendering another <Broadcast>, just change the `value` prop of the one " +
+          "you already rendered."
+      )
+
+      broadcastInstance = this
+
+      if (this.props.value !== currentValue) {
+        // TODO: Publish and warn about the double render
+        // problem if there are existing subscribers? Or
+        // just ignore the discrepancy?
+      }
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (this.props.value !== nextProps.value) {
+        publish(nextProps.value)
```

Comment:
What are the implications of calling `publish` in `componentWillReceiveProps` rather than `componentDidUpdate`?

Reply:
I used cWRP because `publish` *does* call `setState`, just in the `<Subscriber>`s instead of the `<Broadcast>`. This is how [the current `<Broadcast>` implementation works](https://github.com/ReactTraining/react-broadcast/blob/master/modules/Broadcast.js#L69-L75), and we've had good results with it. But something may have changed in Fiber.

Reply:
I'm not familiar with `react-broadcast` so I was just asking 😄  I saw your comment on the context PR requesting feedback on this, so I scanned the diff for potential problems.